bakery/src/scripts/utils.py

```diff
@@ -85,3 +87,42 @@ def parse_uri(uri):
         return None
     legacy_id, legacy_version = uri.split('@')
     return legacy_id, legacy_version
+
+
+def ensure_isoformat(timestamp):
+    """Given a timestsamp string either validate it is already ISO8601 and
+    return or attempt to convert it.
+    """
+    try:
+        # Using dateutil.parser here instead of datetime.fromisoformat as the
+        # former seems to be more robust and avoids some false negatives seen
+        # with the latter (e.g. with "2021-03-22T14:14:33.17588-05:00")
+        parser.isoparse(timestamp)
+        return timestamp
+    except ValueError:
+        # The provided timestamp needs to be converted
+        pass
+
+    # Try with a timezone offset value that we can parse
+    try:
+        return datetime.strptime(timestamp, "%Y/%m/%d %H:%M:%S %z").isoformat()
+    except ValueError:
+        pass
+
+    # Try with timezone strings that we can interpret as offset -0500
+    supported_formats = [
+        "%Y/%m/%d %H:%M:%S.%f GMT-5",
+        "%Y/%m/%d %H:%M:%S GMT-5",
+        "%Y/%m/%d %H:%M:%S.%f US/Central"
+    ]
```

Comment:
If we are interested in being more forgiving with these dates as well, we can parse the timezone separately
```python
from dateutil.parser import parse
from dateutil.tz import gettz
evil_date = '2021/04/20 13:00:00.000 US/Central'
# String splitting isn't perfect either, but I couldn't find a way to get
# dateutil to parse the last part as a timezone alias
evil_date_notz, evil_date_tz = evil_date.rsplit(' ', 1)
parse(evil_date_notz, ignoretz=True).replace(tzinfo=dateutil.tz.gettz(evil_date_tz))
# => datetime.datetime(2021, 4, 20, 13, 0, tzinfo=tzfile('/usr/share/zoneinfo/US/Central'))
```
However, I think this may require some system library to be installed due to the `/usr/share/zoneinfo/US/Central` piece